host/3.0/buster/amd64/base/generate-composite.sh

```diff
@@ -0,0 +1,54 @@
+#! /bin/bash
+
+
+
+DIR=$(dirname $0)
+IFS=$'\n'
+
+globalize_args() {
+    ## This method is used to bring all args preceding any From statements to the top of the file. 
+    ## Allows for ARGs defined globally in composite images to be global in the compiled dockerfile
+    gblarg=($((sed -n '/FROM/q;p' $1) | grep ARG)) # list of args in arr with carriage returns at end
+    printf "\t ARGs globalized for $1 : \n"
+    for arg in ${gblarg[@]};
+    do
+        printf "\t\t $arg \n"
+    done 
+
+}
+
+combine_dockerfile() {
+    printf $"  Combining $base with $1\n"
+    runtime=`echo $1 | cut -d'/' -f 2`
+    runtimeversion=`echo $1 | cut -d'/' -f 3`
+    printf "\t Image Runtime and Version: $runtimeversion \n"
+
+    printf $"\t Collecting global arguments in $1 to globalize them in the final dockerfile.\n"
+    globalize_args $1
+    printf '%s\n' "${gblarg[@]}" > ./temp.txt
```

Comment:
I **think** that I want to not use append mode here as every individual lang/version will overwrite this file.

Reply:
You can do 

```
echo "${gblarg[@]}" | cat - $base $1 > ./out/${runtime}/${runtimeversion}-appservice.Dockerfile
```

To avoid the need for a tmp file.

Reply:
Incorporated this in most recent commit. Thank you both